docs/proposals/20220825-azuremanaged-cluster-exp-graduation.md

```diff
@@ -0,0 +1,73 @@
+
+See:
+
+- https://github.com/kubernetes-sigs/cluster-api/pull/6988
+
+The above proposal defines a set of recommendations for Cluster API providers implementing Managed Kubernetes solutions. Because this proposal is an opt-in collection of architectural recommendations, it is not required that "`AzureManagedCluster`" strictly agrees with everything therein. However, by contributing to the successful landing of that proposal in Cluster API, we can best ensure that any capz-specific opinions or learnings are reflected, for the benefit of the larger ecosystem, and for the maximum happiness of AKS customers in particular.
+
+### 2. Integrate Cluster API Proposal Recommendations
+
+Once the proposal is accepted and merged into the Cluster API project as an endorsed set of provider recommendations, capz can audit the existing "`AzureManagedCluster`" experimental implementation for areas of disagreement with said recommendations. For each discovery of disagreement, there should be a defensible reason to matriculate a divergent capz Managed Kubernetes implementation into a post-experimental definition of "`AzureManagedCluster`", and ideally some form of sign-off from other provider maintainers. Where community consensus cannot be reached, capz should strongly consider evolving the experimental "`AzureManagedCluster`" implementation to meet the Cluster API Managed Kubernetes recommendation as a pre-requisite to graduating from experimental.
+
+### 3. Full AKS Feature Support
+
+"`AzureManagedCluster`" should be able to accommodate the entire feature set of AKS. Rather than require the concrete list of features of AKS to be fully implemented as a prerequisite to graduation, we should instead audit the AKS feature matrix against the "`AzureManagedCluster`" architectural surface area and ensure that capz is well prepared to continually integrate existing and new AKS features into "`AzureManagedCluster`" along a non-breaking path forward.
+
+### 4. Affinity between capz / Cluster API resource lifecycle enforcement and AKS lifecycle enforcement.
+
+Not unrelated to the above audit of AKS features, we should also overlay the Cluster API lifecycle combinatorics (simply speaking: Create, Read, Update, Delete) against the set of AKS cluster primitives to ensure that sane interfaces exist in capz in order to effectively enforce (for example, add an AKS node pool), or passively delegate authority (for example, defer to the built-in AKS autoscaler when enabled to enforce `MachinePool` replica count), if appropriate.
+
+### 5. Azure API Request Optimizations
+
+Prior to graduation from experimental, we should ensure that the sufficient set of configurable interfaces exist to allow "`AzureManagedCluster`" users to effectively tune capz so that no unnecessary Azure API requests are introduced into their AKS environment. We should ship sane, optimized defaults, but allow for flexible overrides to anticipate a wide variety of operational use cases.
```

Comment:
What do you mean by "unnecessary Azure API requests are introduced"? Are you referring to API requests made by capz during reconciliation or to those needed by an external client to configure the cluster?

Reply:
Primarily I mean that capz + AKS offers an opportunity for more AKS API requests from two (that I can think of off the top of my head) dimensions:

1. the implementation details of eventual consistency introduce net new "let me check if the current state matches my goal state" GET calls that aren't there if you aren't running an eventual consistency controller on top of your AKS environment
2. if we do our jobs correctly capz will optimize for multi-cluster stories, which may yield customer scenarios where more clusters are running on single subscriptions, which is at least _n_x (where _n_ is the number of clusters) calls to the AKS API

So basically, we want to be a very well-behaved API citizen to introduce all of this neat foo while still keeping Azure subscriptions actually functional. :)